### DISCUSSION TITLE
Build chained any query for singleValueExtendedProperties

### DISCUSSION THREAD

**Discussion by tobias-reese** [O365/python-o365#1101] [Q&A]
How do i build a query like this?

`
singleValueExtendedProperties/Any(ep: ep/id eq 'String {6ac8ccc0-0bad-4679-ac14-2d0501728d3e} Name MyCustomProperty' and ep/value eq 'myvalue')
`

I tried:
`
q.any(collection='singleValueExtendedProperties', word='String {6ac8ccc0-0bad-4679-ac14-2d0501728d3e} Name MyCustomProperty', operation='eq', attribute='id')
q.chain('and').any(collection='singleValueExtendedProperties', word='myvalue', operation='eq', attribute='value')
`

but this results in
`$filter=singleValueExtendedProperties/any(a:a/id+eq+'String+{6ac8ccc0-0bad-4679-ac14-2d0501728d3e}+Name+MyCustomProperty')+and+singleValueExtendedProperties/any(a:a/value+eq+'myvalue')` 

which leads to

`Error Message: The filter expression for $filter does not match to a single extended property and a value restriction.`

**Comment by tobias-reese** [comment]
I managed to create the desired query, but would like to know if there is a clean way to achieve this.

`
q.any(collection='singleValueExtendedProperties', word='String {6ac8ccc0-0bad-4679-ac14-2d0501728d3e} Name MyCustomProperty\' and a/value eq \'myvalue', operation='eq', attribute='id')
`

Produces:
`
singleValueExtendedProperties/any(a:a/id eq 'String {6ac8ccc0-0bad-4679-ac14-2d0501728d3e} Name MyCustomProperty' and a/value eq 'myvalue')
`

**Comment by alejcas** [comment]
The Query object is there to help. You can always pass a string with your OData filter wherever you pass a `Query` instance.
